persistent/persistent.cabal

```diff
@@ -37,6 +37,7 @@ library
       , template-haskell         >= 2.13 && < 2.18
       , text                     >= 1.2
       , th-lift-instances        >= 0.1.14    && < 0.2
+      , th-abstraction           >= 0.2.8
```

Comment:
It is difficult to maintain code that introspects `Dec` for multiple versions of template-haskell. The fields often get added or moved. This offers a stable abstraction and an alternative to CPP pragmas.